src/Kernel/Extensions.cs

```diff
@@ -23,5 +29,61 @@ public static void AddIQSharpKernel(this IServiceCollection services)
             services.AddSingleton<IExecutionEngine, Kernel.IQSharpEngine>();
             services.AddSingleton<IConfigurationSource, ConfigurationSource>();
         }
+
+        internal static IEnumerable<QsDeclarationAttribute> GetAttributesByName(
+            this OperationInfo operation, string attributeName,
+            string namespaceName = "Microsoft.Quantum.Documentation"
+        ) =>
+            operation.Header.Attributes.Where(
+                attribute =>
+                    attribute.TypeId.Item.Namespace.Value == namespaceName &&
+                    attribute.TypeId.Item.Name.Value == attributeName
+            );
+
+        internal static bool TryAsStringLiteral(this TypedExpression expression, [NotNullWhen(true)] out string value)
```

Comment:
Cool, wasn't aware of these post-condition attributes!

Reply:
C# 8 has some neat features, if a bit hidden!